libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -397,9 +398,9 @@ def similarity_search(
         self, query: str, k: int = 4, **kwargs: Any
     ) -> List[Document]:
         search_type = kwargs.get("search_type", self.search_type)
-        if search_type == "similarity":
+        if search_type in ("similarity", "similarity_score_threshold"):
             docs = self.vector_search(query, k=k, **kwargs)
-        elif search_type == "hybrid":
+        elif search_type in ("hybrid", "hybrid_score_threshold"):
```

Comment:
the `*_score_threshold` search type is now automatically handled passing the `score_threshold` kwarg to the search method. Otherwise, we don't filter the scores (as expected).